content.test.ts

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs';
+import path from 'path';
+
+describe('Codecademy Docs Content', () => {
+  it('adheres to content file structure', () => {
+    // file names can only contain alphanumerics and hyphens
+    const validateName = (pathName: string, name: string) => {
+      const filteredName = name.replace(/[^A-Za-z0-9.-]/g, '');
+
+      // format so that test failures are more helpful
+      if (filteredName !== name) {
+        expect(`${pathName} - file name must only include alphanumerics or -`).toBe('');
+      }
+    };
```

Comment:
This is a little jank! I wanted to avoid iterating through all the files more than once so I ended up rolling a bunch of tests into one. For now the tests actually run quite fast so I can split this test into at least 2 if that seems better.